src/ModuleLoader.ts

```diff
@@ -280,6 +280,7 @@ export class ModuleLoader {
 
 		const module: Module = new Module(this.graph, id, moduleSideEffects, isEntry);
 		this.modulesById.set(id, module);
+		this.graph.watchFiles[id] = true;
```

Comment:
I think this is the best place to add the new watchFile; with this logic, files added via `this.addWatchFile` in a plugin will also not be forgotten.